### PR TITLE
properly shutdown actor system after tests

### PR DESCRIPTION
**Pull request by coreyjonoliver**

### CODE REVIEW

**Comment by bpholt** [comment]
Since this project doesn't use the sbt-release plugin, we either need to convert it to do so, or update the version manually.

**Comment by coreyjonoliver** [comment]
Any opposition to me adding support for the `sbt-release` plugin as part of this pull request?

**Comment by bpholt** [comment]
@coreyjonoliver I'm on board with either approach. Adding sbt-release will take longer and isn't strictly necessary, so I didn't want to require it, but I think it would be worth the effort to do so now that we know how to do it.

File: .travis.yml [hunk 1]
```diff
@@ -1,20 +1,42 @@
...
+
+before_script:
+- jdk_switcher use $JDK
 
 script: sbt ++$TRAVIS_SCALA_VERSION test
 
-before_deploy:
-  - mkdir -p $HOME/.bintray
-  - openssl aes-256-cbc -K $encrypted_ba8b6e0b5d99_key -iv $encrypted_ba8b6e0b5d99_iv -in .credentials.enc -out $HOME/.bintray/.credentials -d
+before_install:
```
Comment (bpholt) [inline .travis.yml]: Why does this change from `before_deploy` to `before_install`?

File: .travis.yml [hunk 2]
```diff
@@ -1,20 +1,42 @@
...
+- openssl aes-256-cbc -K $encrypted_30b060b1770b_key -iv $encrypted_30b060b1770b_iv
```
Comment (bpholt) [inline .travis.yml]: Does splitting this across multiple lines not cause any problems? I'm surprised it works.

File: .travis.yml [hunk 3]
```diff
@@ -1,20 +1,42 @@
...
+  -in .travis/credentials.enc -out .travis/credentials -d
```
Comment (bpholt) [inline .travis.yml]: I think the bintray publisher plugin looks for the credentials in a specific place, which is why they were previously placed in `$HOME/.bintray/.credentials`

File: .travis.yml [hunk 4]
```diff
@@ -1,20 +1,42 @@
 language: scala
 
 scala:
-  - 2.11.8
-  - 2.12.1
+- 2.11.8
```
Comment (bpholt) [inline .travis.yml]: Does this whitespace change make the file parse differently? Not sure if it was intentional or not.

**Comment by bpholt** [comment]
Is Akka only built for Java 8 now? I think that's why the builds are failing on Scala 2.11, because we're running them with Java 7, but that used to work. We're not pinning a specific version of Akka in build.sbt, so maybe something changed upstream?

File: .travis.yml (hunk 3 above)
Comment (coreyjonoliver) [inline .travis.yml]: Gotcha. I'll make a fix for this.

**Comment by coreyjonoliver** [comment]
Yeah. I'm not sure why they would be failing either.

File: .travis.yml (hunk 1 above)
Comment (coreyjonoliver) [inline .travis.yml]: It's what the TravisCI CLI defaults to when you use:

`travis encrypt-file credentials --add`

Comment (bpholt) [inline .travis.yml]: Sure, but it's not the behavior we want, right? (FWIW I don't use `--add` because of stuff like this.)

Comment (coreyjonoliver) [inline .travis.yml]: That's a fair point. I can revert to using `before_deploy`.

**Comment by coreyjonoliver** [comment]
For what it's worth, looking at past builds in TravisCI looks like we always only cross compiled against Java 8.

File: .travis.yml (hunk 4 above)
Comment (coreyjonoliver) [inline .travis.yml]: I don't think it should be parsed differently. The file was reformatted when I originally ran `travis encrypt-file credentials --add`.

**Comment by coreyjonoliver** [comment]
Looks like support for Java 7 was dropped back when 2.4.0 was [released](http://akka.io/news/2015/09/30/akka-2.4.0-released.html):

```
dropped support for Java 6 & 7 as announced in the last roadmap update, and now require Java 8 or later—this will allow us to create modern idiomatic Java APIs over the course of the upcoming releases
```

I'll drop support for Java 7 from the TravisCI file.

File: .travis.yml [hunk 5]
```diff
@@ -1,20 +1,32 @@
...
+  - secure: "Zrqn8sQAqgy7NfVGXwQ6XcKF5riCRQJC9MKzqhT80g+RTX8NfkccjF42xWsxfXjQixpxIItYmFWZDtcgH6yrqJLHS488b+yBNHzg1f6mDfQoPstf01OWWZRLOV25A/wKW2Y8lfKexvTHaUudhZAEkpcfRSKaGLqxqDV+4/iRAxSDeknauWD3V2qi2rUmxVdeFwKk4aPMprxQZj104hDok1nBH580M/Oy3c2fq+0YcSV9qDI4BtzGn6TbDZ0YEOq7/BORlQBLzS6hAnHib2Nn2iUg4+AKb8FFphGNJuIY8jLa0WWmcJY/jkLb+NgoQg9TT0fo0qOoefUABEZsIdiShr2s2nFqroBjucOzdXBqTv9DFJE8M815/wmvVJe4lcukMs3ePlYZ7wrr8pcmNQpuXl4rCklRsAHqbX7F0poi3tlSSb0KQY86Ky2dLDc6hwtU73dhAw/K8MPE4Vvl44F2VkLPd9KXUpBHuWcfYtceS4cF3X3ZRoD4c1nGp+dS3Z9nUne023l+oTPVkvuLPiK3UI1fg4/qktUOTN16SU87VFzIT7yAfipSdukmNoxzfzibTNgefVN7Y5Wvvt6LtTtJhy57M6hccmipyjVpvoJyhgA/tJOLCfAjRRI7RpBtec1MUIFpg8uqy2gFmHxdJat50unloCy/S7hwPgx4FGspvAw="
 
 script: sbt ++$TRAVIS_SCALA_VERSION test
 
 before_deploy:
-  - mkdir -p $HOME/.bintray
-  - openssl aes-256-cbc -K $encrypted_ba8b6e0b5d99_key -iv $encrypted_ba8b6e0b5d99_iv -in .credentials.enc -out $HOME/.bintray/.credentials -d
+- mkdir -p $HOME/.bintray
+- openssl aes-256-cbc -K $encrypted_30b060b1770b_key -iv $encrypted_30b060b1770b_iv
+  -in .travis/credentials.enc -out $HOME/.bintray/.credentials -d
```
Comment (bpholt) [inline .travis.yml]: I think it would be wise to combine this with line 15 so the command isn't split across multiple lines.

Comment (coreyjonoliver) [inline .travis.yml]: Combined.

**Pull request merged**